feat: Add rhel base images

bradbeam opened the pull request:
This enables us to build a Certified Image for RH Partner connect.

We still need to figure out how/where to push these images,
most likely through https://access.redhat.com/RegistryAuthentication

Signed-off-by: Brad Beam <brad.beam@stormforge.io>

bradbeam committed:
feat: Add rhel base images

This enables us to build a Certified Image for RH Partner connect.

We still need to figure out how/where to push these images,
most likely through https://access.redhat.com/RegistryAuthentication

Signed-off-by: Brad Beam <brad.beam@stormforge.io>

jgustie commented on Dockerfile.rhel:
```diff
@@ -0,0 +1,19 @@
+      version="latest" \
+      release="1" \
+      summary="Stormforge Optimize uses machine learning to improve the performance of your application" \
+      description="Stormforge Optimize uses machine learning to improve the performance of your application."
```
StormFrog. I mean. StormForge.

bradbeam committed:
feat: Consolidate controller image build

This makes use of BUILD_IMAGE build arg to flex on the base image used
and multistage docker build to flex on the base distro for our nonroot
user.

Signed-off-by: Brad Beam <brad.beam@stormforge.io>

jgustie commented on Dockerfile:
```diff
@@ -1,8 +1,29 @@
+# === Distroless
+FROM  base as distroless
+
+USER nonroot:nonroot
```
I'm not aware of any requirement to use `nonroot`, would it be easier if we always just ran as `nobody`?

bradbeam committed:
feat: Use `nobody` for both controller images

Distroless provides both `nobody` and `nonroot` so this allows
us to simplify the container build.

Signed-off-by: Brad Beam <brad.beam@stormforge.io>
